### ISSUE TITLE
Decide whether to split out OMK or fully fold in

### ISSUE THREAD

**Issue by cmungall** [INCATools/ontology-access-kit#172]
OAK contains a "secret" application, the Ontology Mapping Kit

Code:

https://github.com/INCATools/ontology-access-kit/tree/main/src/oaklib/omk

Currently there is one core method, for performing cross-ontology diffs. E.g. use MP and HP mappings to give a report on how these ontologies are structurally different.

The output conforms to this datamodel:

https://github.com/INCATools/ontology-access-kit/blob/main/src/oaklib/datamodels/cross_ontology_diff.yaml

Tests:

https://github.com/INCATools/ontology-access-kit/tree/main/tests/test_omk

There is also a CLI, with only one command:

```
poetry run omk mdiff --help
Usage: omk mdiff [OPTIONS]

  Calculates a relational diff between ontologies in two sources using the
  combined mappings from both

  E.g. use MP and HP mappings to give a report on how these ontologies are
  structurally different.

Options:
  -S, --source TEXT        ontology prefixes  e.g. HP, MP
  --other-input TEXT       Additional input file
  --other-input-type TEXT  Type of additional input file
  --help                   Show this message and exit.
```

For context, we also have other mapping tools that layer over OAK such as https://github.com/monarch-initiative/mapping-walker/

Should we

1. Properly split this out into a separate repo that depends on OAK
2. Make it a proper part of OAK, and if so
    - make an interface for doing cross-ontology structural diffs, or fold this into existing differ interface
    - make this a separate non-OO utility under utilities

cc @matentzn @joeflack4 @hrshdhgd

**Comment by cthoyt** [comment]
lol i was thinking about this problem and writing my own solution for it recently

**Comment by joeflack4** [comment]
In this case, I favor (2a). My reasoning:

2 > 1: it looks like OMK doesn't do writes; only reads. Therefore it seems like it falls well within the subset of what I imagine OAK is designed to do: idempotent analytical operations on ontologies.

2a > 2b: I think that this is a specific enough case that it merits its own submodule. At the point at which something has its own CLI, I think that is indicative that this shouldn't be in a 'utils' file. Typically I find that a 'utils' file is typically for small functions to make the codebase DRYer as opposed to anything sophisticated / user-facing.

**Comment by cmungall** [comment]
@joeflack4 

> it looks like OMK doesn't do writes; only reads

You are correct in that the focus of OAK is on access, with tools like ROBOT for writes and linkml-owl for generation. But note that starting with [v0.1.27](https://github.com/INCATools/ontology-access-kit/releases/tag/v0.1.27) OAK has partial support for KGCL, which allows a caller to provide "patches" to update an ontology

So overall it seems there is no burning desire to move the functionality out of OAK, so for now let's make it a command alongside all the others.

Good point about `utilities` Joe. I think the fundamental dichotomy here is whether this deserves its own interface (following the OAK pattern for interfaces), which would allow different implementations to implement their own (possibly more performant) ways of doing this, vs whether a simpler "procedural" non-OO approach is better.

For now, I think we should get rid of "omk" as a top-level package, it can go alongside "mappings" in other utilities, this doesn't prohibit us making an interface later (and in fact some implementations already using utility packages, particularly those in graphs)
